lib/pure/parsecfg.nim

```diff
@@ -295,252 +371,440 @@ proc getEscapedChar(c: var CfgParser, tok: var Token) =
+    rawGetTok(c)
   of tkEof:
     result = CfgEvent(kind: cfgEof)
-  of tkDashDash:
-    rawGetTok(c, c.tok)
-    result = getKeyValPair(c, cfgOption)
+  of tkDash, tkDashDash:
+    c.tok.kind = tkSymbol
+    result = getKeyValuePair(c, cfgOption)
+    c.tok.kind = tkInvalid
+    rawGetTok(c)
   of tkSymbol:
-    result = getKeyValPair(c, cfgKeyValuePair)
+    result = getKeyValuePair(c, cfgKeyValuePair)
+    c.tok.kind = tkInvalid
+    rawGetTok(c)
   of tkBracketLe:
-    rawGetTok(c, c.tok)
-    if c.tok.kind == tkSymbol:
-      result = CfgEvent(kind: cfgSectionStart, section: c.tok.literal)
-    else:
-      result = CfgEvent(kind: cfgError,
-        msg: errorStr(c, "symbol expected, but found: " & c.tok.literal))
-    rawGetTok(c, c.tok)
-    if c.tok.kind == tkBracketRi:
-      rawGetTok(c, c.tok)
-    else:
-      result = CfgEvent(kind: cfgError,
-        msg: errorStr(c, "']' expected, but found: " & c.tok.literal))
+    result = getSection(c, cfgSectionStart)
+    c.tok.kind = tkInvalid
+    rawGetTok(c)
   of tkInvalid, tkEquals, tkColon, tkBracketRi:
     result = CfgEvent(kind: cfgError,
       msg: errorStr(c, "invalid token: " & c.tok.literal))
-    rawGetTok(c, c.tok)
+    c.tok.kind = tkInvalid
+    rawGetTok(c)
 
 # ---------------- Configuration file related operations ----------------
 type
-  Config* = OrderedTableRef[string, <//>OrderedTableRef[string, string]]
+  Config* = OrderedTableRef[string, SectionItem]
+
+  SectionItem = tuple
+    sectionRelated: SectionRelated
+    keyValue: <//>OrderedTableRef[string, KeyValueItem]
+
+  KeyValueItem = tuple
+    value: string
+    keyValueRelated: KeyValueRelated
 
-proc newConfig*(): Config =
+proc newConfig*(): <//>Config =
   ## Creates a new configuration table.
   ## Useful when wanting to create a configuration file.
-  result = newOrderedTable[string, <//>OrderedTableRef[string, string]]()
+  result = newOrderedTable[string, SectionItem]()
 
-proc loadConfig*(stream: Stream, filename: string = "[stream]"): <//>Config =
+proc loadConfig*(stream: Stream, filename: string = "[stream]",
+                 commentSymbol = "#;"): <//>Config =
   ## Loads the specified configuration from stream into a new Config instance.
   ## `filename` parameter is only used for nicer error messages.
-  var dict = newOrderedTable[string, <//>OrderedTableRef[string, string]]()
+  ## `commentSymbol` default value is `"#;"`
+  var dict = newOrderedTable[string, SectionItem]()
   var curSection = "" ## Current section,
                       ## the default value of the current section is "",
                       ## which means that the current section is a common
   var p: CfgParser
   open(p, stream, filename)
+  p.commentSymbol = commentSymbol
   while true:
     var e = next(p)
     case e.kind
     of cfgEof:
       break
     of cfgSectionStart: # Only look for the first time the Section
+      var tp: SectionItem
+      var t = <//>OrderedTableRef[string, KeyValueItem]()
```

Comment:
I think theres other PR that removes `<//>`, I think is Deprecated or almost deprecated.